File: java/oop/RectangleApp/src/com/techlabs/rectangle/Rectangle.java
package com.techlabs.rectangle;

public class Rectangle {
	public int height=1;
	public int width=1;
	public String color="no_color";
	
	public Rectangle(){
		System.out.println("Object Created....!!!");
	}
	
	public void setHeight(int height){
		this.height=validate(height);
	}
	
	public int getHeight(){
		return this.height;
	}
	
	public void setWidth(int width){
		this.width=validate(width);
	}
	
	public int getWidth(){
		return this.width;
	}
	public int calculateArea(){
		return height*width;
	}
	
	public void setBorderColor(String color){
		if(color.equalsIgnoreCase("red") || color.equalsIgnoreCase("blue") || color.equalsIgnoreCase("green"))
			this.color=color;
	}
	
	public String getBorderColor(){
		return this.color;
	}
	
	private int validate(int dimension){
		if(dimension<1)
			dimension=1;
		if(dimension>100)
			dimension=100;
		return dimension;
	}
}

File: dotNet/CSharp/RectangleApp/RectangleApp/Rectangle.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RectangleApp
{
    public class Rectangle
    {
        private int _height = 1;
        private int _width = 1;
        private string _color = "No_Color";

        public Rectangle()
        {
            Console.WriteLine("Object Created.");
        }

        public Int32 Height
        {
            get
            {
                return this._height;
            }

            set
            {
                this._height = Validate(value);
            }
        }

        public Int32 Width
        {
            get
            {
                return this._width;
            }
            set
            {
                this._width = Validate(value);
            }
        }

        public Int32 CalculateArea()
        {
            return Width * Height;
        }

        public String Color
        {
            get
            {
                return this._color;
            }
            set
            {
                if (value.Equals("red") || value.Equals("blue") || value.Equals("green"))
                {
                    this._color = value;
                }
            }
        }

        private int Validate(int dimension)
        {
            if (dimension<1)
            {
                dimension = 1;
            }
            else if(dimension>100)
            {
                dimension = 100;
            }
            return dimension;
        }

        public override string ToString()
        {
            return "Rectangle Height= "+Height+" Width= "+Width+" Color="+Color;
        }
    }
}